pkg/controller/jobframework/reconciler_test.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package jobframework
+
+import (
+	"context"
+	"fmt"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	kubeflow "github.com/kubeflow/mpi-operator/pkg/apis/kubeflow/v2beta1"
+	batchv1 "k8s.io/api/batch/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"sigs.k8s.io/kueue/pkg/util/pointer"
+	utiltesting "sigs.k8s.io/kueue/pkg/util/testing"
+	testingjob "sigs.k8s.io/kueue/pkg/util/testingjobs/job"
+	testingmpijob "sigs.k8s.io/kueue/pkg/util/testingjobs/mpijob"
+)
+
+func TestIsParentJobManaged(t *testing.T) {
+	parentJobName := "test-job-parent"
+	childJobName := "test-job-child"
+	jobNamespace := "default"
+	unknwonWlOwner := &metav1.OwnerReference{
+		APIVersion: batchv1.SchemeGroupVersion.String(),
+		Kind:       "Job",
+		Name:       parentJobName,
+		UID:        types.UID(parentJobName),
+		Controller: pointer.Bool(true),
+	}
+
+	cases := map[string]struct {
+		parentJob client.Object
+		job       client.Object
+		want      bool
+		wantErr   error
+	}{
+		"child job doesn't have ownerReference": {
+			parentJob: testingmpijob.MakeMPIJob(parentJobName, jobNamespace).
+				UID(parentJobName).
+				Obj(),
+			job: testingjob.MakeJob(childJobName, jobNamespace).
+				Obj(),
+			wantErr: errChildJobOwnerNotFound,
+		},
+		"child job has ownerReference with unknown workload owner": {
+			parentJob: testingjob.MakeJob(parentJobName, jobNamespace).
+				UID(parentJobName).
+				Obj(),
+			job: testingjob.MakeJob(childJobName, jobNamespace).
+				OwnerReference(parentJobName, batchv1.SchemeGroupVersion.WithKind("Job")).
+				Obj(),
+			wantErr: fmt.Errorf("workload owner %s: %w", unknwonWlOwner.String(), errUnknownWorkloadOwner),
+		},
```

Comment:
@alculquicondor For example, we create a new error object in this case. So we faced the compare error.
Hence, I guess that `cmp.Diff` doesn't work well...

Reply:
So if we use `wantErr:  errUnknownWorkloadOwner`, that should work well.

Reply:
Ah, I see what your suggestion means :)